libraries/adaptive-expressions/src/builtinFunctions/addOrdinal.ts

```diff
@@ -28,7 +28,7 @@ export class AddOrdinal extends ExpressionEvaluator {
      */
     private static evaluator(): EvaluateExpressionDelegate {
         return FunctionUtils.apply(
-            (args: any[]): string => AddOrdinal.evalAddOrdinal(args[0]),
+            (args: readonly unknown[]): string => AddOrdinal.evalAddOrdinal(args[0] as number),
```

Comment:
The goal with `unknown` is to force us to do type checking, not using `as` in its place.

Reply:
Hi.
Yes, but there are some gaps.
In old Typescript versions(3.5.3 here of adaptive-expressions and botbuilder-lg). Automatic type recognition in some case is not intelligent.
For example:
```
let list: unknown[];
if (typeof list[0] === 'string') {
    const result = list[0].toLowerCase();
}

```
It would pass the diagnostic check for the latest Typescript. But would fail in TS 3.5.
The error messages is: 
`Property 'toLowerCase' does not exist on type 'unknown'`

So, there are some solutions to solve this issue, 
First one is to add `as` for the item.
`(list[0] as string).toLowerCase()`
The second one is to assign each item to a const object.
```
let list: unknown[];
const firstChild = list[0];
if (typeof firstChild === 'string') {
    const result = firstChild.toLowerCase();
}

```
What do you think?